pkg/controller/podgc/gc_controller.go

```diff
@@ -249,12 +249,12 @@ func (gcc *PodGCController) gcOrphaned(ctx context.Context, pods []*v1.Pod, node
 			continue
 		}
 		logger.V(2).Info("Found orphaned Pod assigned to the Node, deleting", "pod", klog.KObj(pod), "node", klog.KRef("", pod.Spec.NodeName))
-		condition := corev1apply.PodCondition().
-			WithType(v1.DisruptionTarget).
-			WithStatus(v1.ConditionTrue).
-			WithReason("DeletionByPodGC").
-			WithMessage("PodGC: node no longer exists").
-			WithLastTransitionTime(metav1.Now())
+		condition := &v1.PodCondition{
+			Type:    v1.DisruptionTarget,
+			Status:  v1.ConditionTrue,
+			Reason:  "DeletionByPodGC",
+			Message: "PodGC: node no longer exists",
+		}
```

Comment:
is this going to be changed back to SSA in the future? 

Are the other places that use SSA affected as well?

https://github.com/kubernetes/kubernetes/blob/0851995a61cb83645734183cb49250e0fc3b14a4/pkg/controller/disruption/disruption.go#L776

https://github.com/kubernetes/kubernetes/blob/0851995a61cb83645734183cb49250e0fc3b14a4/pkg/scheduler/framework/preemption/preemption.go#L366

Reply:
There is also this place: https://github.com/kubernetes/kubernetes/blob/0851995a61cb83645734183cb49250e0fc3b14a4/pkg/controller/nodelifecycle/scheduler/taint_manager.go#L132.

I'm thinking about minimal fix for the issue reported by users. Certainly, in this PR it is going to be only PodGC. We may extend the scope based on user feedback or reviewers in follow up PRs. My hope is that after SSA is fixed we can revert this change. FYI @alculquicondor @liggitt for suggestion if this is the right approach.

Reply:
all of the places that apply this condition using SSA and currently short-circuit their deletion if it fails have to be fixed and backported, or we have to disable the PodDisruptionConditions feature gate to restore them functioning properly

Reply:
> Certainly, in this PR it is going to be only PodGC. We may extend the scope based on user feedback or reviewers in follow up PRs

I don't see the benefit of splitting the regression fix across multiple PRs. I think we should switch from SSA to SMP using the method in this PR in all places.

Reply:
Right, any delete operation is affected. Initially I was thinking that only deletions with gracePeriod=0 were affected, but that's not true.

I agree that we should fix in one PR.